components/usage/pkg/apiv1/billing.go

```diff
@@ -67,6 +67,37 @@ func (s *BillingService) UpdateInvoices(ctx context.Context, in *v1.UpdateInvoic
 	return &v1.UpdateInvoicesResponse{}, nil
 }
 
+func (s *BillingService) ReconcileInvoices(ctx context.Context, in *v1.ReconcileInvoicesRequest) (*v1.ReconcileInvoicesResponse, error) {
+	balances, err := db.ListBalance(ctx, s.conn)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to reconcile invoices.")
+		return nil, status.Errorf(codes.Internal, "Failed to reconcile invoices.")
+	}
+
+	creditSummaryForTeams := map[string]stripe.CreditSummary{}
+	for _, balance := range balances {
+		entity, id := balance.AttributionID.Values()
+
+		// TODO: Support updating of user attribution IDs
+		if entity != db.AttributionEntity_Team {
+			continue
+		}
+
+		creditSummaryForTeams[id] = stripe.CreditSummary{
+			Credits:  int64(math.Ceil(balance.CreditCents.ToCredits())),
+			ReportID: "no-report",
+		}
+	}
+
+	err = s.stripeClient.UpdateUsage(ctx, creditSummaryForTeams)
```

Comment:
If I read correctly [here](https://github.com/gitpod-io/gitpod/blob/34ef9cb2edb63a102d881ea48bb1c330e6a235ba/components/usage/pkg/stripe/stripe.go#L83-L86), we might fail to update if the very first query is for a costCenter without stripe account, no?

Reply:
We fail if we fail to lookup customers from stripe. But we then iterate over the stripe customers which actually exist rather than over all the attribution IDs so in this case it would not fail on the first which doesn't have a stripe account.